src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs

```diff
@@ -431,20 +487,24 @@ internal void RefreshCaches(ReadOnlySpan<char> path)
             // Check if the main path is a directory, or a link to a directory.
             int fileType = _fileCache.Mode & Interop.Sys.FileTypes.S_IFMT;
             bool isDirectory = fileType == Interop.Sys.FileTypes.S_IFDIR ||
-                               (fileType == Interop.Sys.FileTypes.S_IFLNK
-                                && Interop.Sys.Stat(path, out Interop.Sys.FileStatus target) == 0
-                                && (target.Mode & Interop.Sys.FileTypes.S_IFMT) == Interop.Sys.FileTypes.S_IFDIR);
+                               (handle is null && // Don't follow links for SafeHandle APIs.
```

Comment:
> Don't follow links for SafeHandle APIs.

The comment should be improved..

This is an optimization to avoid the stat call.
We don't need to follow links for SafeHandle APIs as they have already been followed when then handle was opened.

> I assume you can have a SFH related to a directory as well.

Yes, though .NET doesn't have APIs to open them.

> Also, this is inconsistent with the setattrlist code.

Here the comment is valid because we're calling `setattrlist` with a `string` even when the user called an API with a `SafeHandle`. But that will change if we're calling `fsetattrlist` instead.